examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/AtpResource.java

```diff
@@ -50,10 +46,15 @@
     private static final Logger LOGGER = Logger.getLogger(AtpResource.class.getName());
 
     private final OciAutonomousDb autonomousDb;
+    private final PoolDataSource atpDataSource;
+    private final String atpServiceName;
 
     @Inject
-    AtpResource(OciAutonomousDb autonomousDb) {
+    AtpResource(OciAutonomousDb autonomousDb, @Named("atp") PoolDataSource atpDataSource,
+                @ConfigProperty(name = "oracle.ucp.jdbc.PoolDataSource.atp.serviceName") String atpServiceName) {
```

Comment:
I'm reading this maybe without context, but now your datasource name _has_ to be `atp` in configuration or this won't work.  In JPA (which I know is not what you're doing), there's an extra layer of indirection: the `persistence.xml` names the datasource, and then that name is used to look up all the JDBC configuration.  You might consider something similar here: maybe have a (meta-, really) configuration item named something like `dataSourceName` whose default value is `atp`, and then use MicroProfile Config programmatically to look up what you need using the value of that property as necessary.

Reply:
User can name it to their liking. I just used "atp" in this example's config so using that.

Reply:
This is example code/resource, we don't rely on "atp" datasource config in our code.